internal/services/price_service.go

```diff
@@ -106,6 +108,37 @@ func (p *PriceResolver) GetCoinGeckoPrice(priceProviderId string, currency strin
 	return result[priceProviderId][currency], nil
 }
 
+func (p *PriceResolver) GetLiFiPrice(chain, contractAddress string) (float64, error) {
```

Comment:
In case you wanted to use thorwallet api as a backup here is an example (unsure which price is more accurate):

`https://api-v2-prod.thorwallet.org/price/ETH.VTHOR-0X815C23ECA83261B6EC689B60CC4A58B54BC24D8D`

returns
```
{
  "ETH.VTHOR-0X815C23ECA83261B6EC689B60CC4A58B54BC24D8D": {
    "usd": "0.44619487"
  }
}
```

```
type PriceResolver struct {
	logger          *logrus.Logger
	cmcMap          *CmcMapResp
	lifiBaseAddress string
	thorWalletBaseAddress string
}

func NewPriceResolver() (*PriceResolver, error) {
	pr := &PriceResolver{
		logger:          logrus.WithField("module", "price_resolver").Logger,
		lifiBaseAddress: "https://li.quest",
		thorWalletBaseAddress: "https://api-v2-prod.thorwallet.org",
	}
	result, err := pr.getCMCMap()
	if err != nil {
		return nil, fmt.Errorf("fail to get CMC map,err: %w", err)
	}
	pr.cmcMap = result
	return pr, nil
}
```
```
func (p *PriceResolver) GetThorwalletPrice(chain, token, contractAddress string) (float64, error) {
		url := fmt.Sprintf("%s/price/%s.%s-%s", p.thorWalletBaseAddress, chain, token, contractAddress)
		resp, err := http.Get(url)
		if err != nil {
				p.logger.Error(err)
				return 0, fmt.Errorf("fail to get price from ThorWallet,err: %w", err)
		}
		defer p.closer(resp.Body)

		if resp.StatusCode != http.StatusOK {
				return 0, fmt.Errorf("error fetching ThorWallet price: %s", resp.Status)
		}

		var result map[string]map[string]string
		if err := json.NewDecoder(resp.Body).Decode(&result); err != nil {
				p.logger.Error("Error parsing JSON:", err)
				return 0, fmt.Errorf("error decoding ThorWallet price response: %w", err)
		}

		// Extract the price from the nested structure
		resultKey := chain + "." + token + "-" + contractAddress
		strPrice := result[resultKey]["usd"]
		price, err := strconv.ParseFloat(strPrice, 64)
		if err != nil {
				return 0, fmt.Errorf("error parsing price: %w", err)
		}

		return price, nil
}
```

```
price, err := pr.GetThorwalletPrice("ETH","VTHOR","0X815C23ECA83261B6EC689B60CC4A58B54BC24D8D")
```

Reply:
thanks @SxMShaDoW , for now , get from li.fi should be ok , if Li.Fi doesn't work , then will switch to thorwallet api